### DISCUSSION TITLE
Kiali using flux and helm operator

### DISCUSSION THREAD

**Discussion by rajivchirania** [kiali/kiali#3695] [Q&A]
I am trying to install kiali operator using flux which needs a helm release.
However the pod keeps failing to start with the below error:

flag provided but not defined: -zap-level

Usage of /opt/kiali/kiali:
  -alsologtostderr
    	log to standard error as well as files
  -config string
    	Path to the YAML configuration file. If not specified, environment variables will be used for configuration.
  -log_backtrace_at value
    	when logging hits line file:N, emit a stack trace
  -log_dir string
    	If non-empty, write log files in this directory
  -logtostderr
    	log to standard error instead of files
  -stderrthreshold value
    	logs at or above this threshold go to stderr
  -v value
    	log level for V logs
  -vmodule value
    	comma-separated list of pattern=N settings for file-filtered logging

Below is my helm release file:

apiVersion: helm.fluxcd.io/v1
kind: HelmRelease
metadata:
name: kiali-operator
namespace: services
annotations:
fluxcd.io/automated: "false"
spec:
helmVersion: v3
releaseName: kiali-operator
chart:
git: ssh://blabla@bla.com
path: bla
ref: bla
rollback:
enable: false
wait: true
timeout: 600
values:
replicaCount: 1
metrics:
enabled: true
debug:
enabled: true
verbosity: "1"
image:
repo: quay.io/kiali/kiali
tag: "v1.22.0"
pullPolicy: IfNotPresent
pullSecrets: []

watchNamespace: ""

clusterRoleCreator: true

cr:
  create: true
  name: kiali
  namespace: services
  spec:
    istio_component_namespaces: "istio-system"
    image_version: "v1.22.0"
    strategy: "anonymous"
    deployment:
      accessible_namespaces:
      - "**"
    ingress_enabled: true
    override_ingress_yaml:
      metadata:
        annotations:
          kubernetes.io/ingress.class: "nginx"
          cert-manager.io/cluster-issuer: "letsencrypt-production"
          cert-manager.io/acme-challenge-type: "dns01"
          cert-manager.io/acme-dns01-provider: "route53"
      spec:
        rules:
        - http:
            paths:
            - path: /kiali
              backend:
                serviceName: kiali
                servicePort: 20001%
I am running istio version 1.6.0
Can someone help what i am doing wrong..

**Comment by jmazzitelli** [comment]
> repo: quay.io/kiali/kiali

Wrong image - you are telling the operator to start the kiali server container image - that's wrong.

There should be no need to specify the image - the default that will be used by the operator helm chart is correct. You overrode the default to a wrong image.

**Reply by rajivchirania** [reply]
But what i read is istio 1.6.0 can support upto kiali 1.22.0
So can you make me understand. which portion i have to remove from the code

**Reply by rajivchirania** [reply]
And even if remove the image_version from the code it still end up with the same error code

flag provided but not defined: -zap-level
Usage of /opt/kiali/kiali:
  -alsologtostderr
    	log to standard error as well as files
  -config string
    	Path to the YAML configuration file. If not specified, environment variables will be used for configuration.
  -log_backtrace_at value
    	when logging hits line file:N, emit a stack trace
  -log_dir string
    	If non-empty, write log files in this directory
  -logtostderr
    	log to standard error instead of files
  -stderrthreshold value
    	logs at or above this threshold go to stderr
  -v value
    	log level for V logs
  -vmodule value
    	comma-separated list of pattern=N settings for file-filtered logging

**Comment by jmazzitelli** [comment]
> repo: quay.io/kiali/kiali

This is wrong. You are telling the operator helm chart to use the container image for Kiali server. You should not have to specify this (unless you desire to pull in images from your own repo - but in this case, it looks like you just want the quay.io images, so there is no need to specify `repo`).

If you insist on setting the repo setting yourself, you must tell it to use the kiali-operator image: `repo: quay.io/kiali/kiali-operator`

**Reply by rajivchirania** [reply]
I made the changes to point to the correct repo so now the operator pod is able to start but kiali pod is getting error..

flag provided but not defined: -v
Usage of /opt/kiali/kiali:
  -config string
    	Path to the YAML configuration file. If not specified, environment variables will be used for configuration.

**Reply by jmazzitelli** [reply]
> flag provided but not defined: -v

That flag was removed in a previous release (see [the commit](https://github.com/kiali/kiali-operator/commit/a05fec1508d309c424bbe1e04f580dc384db6087) that went into v1.27) - see [this change here](https://github.com/kiali/kiali-operator/pull/187/files#diff-bc30d1d38d9f4fd8fe0ef8affe36976a11b729129db008c0635738e23ff13746L63-L64). which was part of this issue: https://github.com/kiali/kiali/issues/2893

So you can't override `image_version` to v1.22 when using the latest version of the operator (because the latest operator will use a Deployment that is incompatible with that version of the server). This is why we put [the caveat in our docs regarding image_version](https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L254-L255): `If you set this to a specific version (i.e. you do not leave it as the default empty string), you must make sure that image is supported by the operator.`

You can either (a) move to image_version v1.27 or (b) use the v1.22 operator helm chart which should use the v1.22 operator (and hence have the old Deployment which works with the v1.22 server).

**Reply by rajivchirania** [reply]
I am using the helm chart version v1.22.0 because thats the highest compatible version with the istio 1.6.0
If i keep the image_version:"" blank or override it to v1.22  i get the same error
So my question is can i use image_version v1.27 with helm chart v1.22.0

**Reply by jmazzitelli** [reply]
> If i keep the image_version:"" blank or override it to v1.22 i get the same error

I suspect you are doing something incorrectly. Post the command you used to run helm and what is your values file/options (and wrap any yaml you post in triple-tic-marks so github formats it for easier reading:

\`\`\`yaml
post: yaml like this
\`\`\`

**Reply by rajivchirania** [reply]
Thanks for your help..I figured it out what mistake i did..Now it works fine

**Reply by jshaughn** [reply]
If you think it may help others, please feel free to briefly describe how you solved the problem.  Thanks!
